### ISSUE TITLE
how to support fp16 in transducer_loss.py 

### ISSUE THREAD

**Issue by jinggaizi** [speechbrain/speechbrain#779]
hi,
i saw the transducer loss in transducer_loss.py, it's defined float32, if i use torch.amp, maybe the loss computation does not support fp16, how can i modify to support fp16 while computing transducer loss

**Comment by TParcollet** [comment]
@aheba @lorenlugosch I believe that this question is for you guys.

**Comment by aheba** [comment]
I'm on it, let me do a PR for that

**Comment by songtaoshi** [comment]
@jinggaizi Hi, have you ever trained successfully using transducer_loss.py ? If success, what is the lr ,dataset, batch_size, decoding method

**Comment by songtaoshi** [comment]
@aheba @lorenlugosch Hi, Have you trained successfully on any dataset? I do not see any report about the RNNT CER on any dataset.    FYI @TParcollet

**Comment by lorenlugosch** [comment]
PER for a transducer recipe is reported for TIMIT here: https://github.com/speechbrain/speechbrain/tree/develop/recipes/TIMIT/ASR/transducer

Don't know if we have any other transducer recipes

**Comment by lorenlugosch** [comment]
Ah, we also have a LibriSpeech transducer recipe, but it doesn't work very well: https://github.com/speechbrain/speechbrain/tree/develop/recipes/LibriSpeech/ASR/transducer

**Comment by tbright17** [comment]
> I'm on it, let me do a PR for that

Hi, just want to check if you are going to make a PR about transducer loss in half precision. Is there any specific plan?

Thanks

**Comment by mravanelli** [comment]
@aheba

**Comment by TParcollet** [comment]
@tbright17 PyTorch is working on the transducer loss, and we'll switch to it as soon as they give it to community.

**Comment by Gastron** [comment]
Closing as the issue was answered.
